### ISSUE TITLE
Custom database action script does not get updated in tfstate

### ISSUE THREAD

**Issue by Kos** [auth0/terraform-provider-auth0#108]
### Describe the problem

Given a database connection with custom database scripts,
when I update the login script within Auth0 dashboard
and I run `terraform plan`, 
then Terraform state is not refreshed
and Terraform does not show the mismatch between production state and expected state.

### What was the expected behavior?

I'd expect Terraform to notice that the database script has changed (compared to the tfstate), refresh the tfstate, and show a plan that will reconcile it with the resource definition. 

### Reproduction

1. Create a database connection with custom script
    ```
    resource "auth0_connection" "sample_connection" {
      name     = "sample_connection"
      strategy = "auth0"
      options {
        enabled_database_customization = "true"
        brute_force_protection         = true
        import_mode                    = false
        disable_signup                 = true
        requires_username              = false
        custom_scripts = {
          login           = "// Sample login script - version from TF"
        }
      }
    }

    ```
2. `terraform apply` to create connection in auth0
3. open Auth0 web UI, change a line in the custom script, save
4. run `terraform plan`, observe output (TF claims there are no changes)

### Environment

- `terraform-provider-auth0` version `0.28.1`

**Comment by sergiughf** [comment]
Hey @Kos appreciate you made us aware of this issue. We'll take a look soon at it.

**Comment by sergiughf** [comment]
Hey @Kos appreciate you made us aware of this issue. We'll take a look soon at it.

**Comment by sergiughf** [comment]
Hey @Kos, apologies for the late reply. I tried to reproduce this but wasn't able. Here's what I did:

## Reproduction

1. Run `terraform apply` with the following:
```hcl 
resource "auth0_connection" "sample_connection" {
    name     = "sample-connection-issue108"
    strategy = "auth0"
    options {
        enabled_database_customization = "true"
        brute_force_protection         = true
        import_mode                    = false
        disable_signup                 = true
        requires_username              = false
        custom_scripts                 = {
            get_user = <<EOF
function getByEmail (email, callback) {
  return callback(new Error("Whoops!"))
}
EOF
        }
    }
}
```

2. Go to **Auth0 management dashboard** -> Authentication -> Database -> Click on "sample-connection-issue108" -> Custom Database -> Database Action Scripts -> Get User
3. Modify script and hit save.
4. Run `terraform plan` and check output matches:

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # auth0_connection.sample_connection will be updated in-place
  ~ resource "auth0_connection" "sample_connection" {
        id                   = "con_XXXXXXXX"
        name                 = "sample-connection-issue108"
        # (5 unchanged attributes hidden)

      ~ options {
          ~ custom_scripts                 = {
              - "change_password" = <<-EOT
                    function changePassword(email, newPassword, callback) {
                      // This script should change the password stored for the current user in your
                      // database. It is executed when the user clicks on the confirmation link
                      // after a reset password request.
                      // The content and behavior of password confirmation emails can be customized
                      // here: https://manage.auth0.com/#/emails
                      // The `newPassword` parameter of this function is in plain text. It must be
                      // hashed/salted to match whatever is stored in your database.
                      //
                      // There are three ways that this script can finish:
                      // 1. The user's password was updated successfully:
                      //     callback(null, true);
                      // 2. The user's password was not updated:
                      //     callback(null, false);
                      // 3. Something went wrong while trying to reach your database:
                      //     callback(new Error("my error message"));
                      //
                      // If an error is returned, it will be passed to the query string of the page
                      // where the user is being redirected to after clicking the confirmation link.
                      // For example, returning `callback(new Error("error"))` and redirecting to
                      // https://example.com would redirect to the following URL:
                      //     https://example.com?email=alice%40example.com&message=error&success=false

                      const msg = 'Please implement the Change Password script for this database ' +
                        'connection at https://manage.auth0.com/#/connections/database';
                      return callback(new Error(msg));
                    }
                EOT -> null
              - "create"          = <<-EOT
                    function create(user, callback) {
                      // This script should create a user entry in your existing database. It will
                      // be executed when a user attempts to sign up, or when a user is created
                      // through the Auth0 dashboard or API.
                      // When this script has finished executing, the Login script will be
                      // executed immediately afterwards, to verify that the user was created
                      // successfully.
                      //
                      // The user object will always contain the following properties:
                      // * email: the user's email
                      // * password: the password entered by the user, in plain text
                      // * tenant: the name of this Auth0 account
                      // * client_id: the client ID of the application where the user signed up, or
                      //              API key if created through the API or Auth0 dashboard
                      // * connection: the name of this database connection
                      //
                      // There are three ways this script can finish:
                      // 1. A user was successfully created
                      //     callback(null);
                      // 2. This user already exists in your database
                      //     callback(new ValidationError("user_exists", "my error message"));
                      // 3. Something went wrong while trying to reach your database
                      //     callback(new Error("my error message"));

                      const msg = 'Please implement the Create script for this database connection ' +
                        'at https://manage.auth0.com/#/connections/database';
                      return callback(new Error(msg));
                    }
                EOT -> null
              - "delete"          = <<-EOT
                    function remove(id, callback) {
                      // This script remove a user from your existing database.
                      // It is executed whenever a user is deleted from the API or Auth0 dashboard.
                      //
                      // There are two ways that this script can finish:
                      // 1. The user was removed successfully:
                      //     callback(null);
                      // 2. Something went wrong while trying to reach your database:
                      //     callback(new Error("my error message"));

                      const msg = 'Please implement the Delete script for this database ' +
                        'connection at https://manage.auth0.com/#/connections/database';
                      return callback(new Error(msg));
                    }
                EOT -> null
              ~ "get_user"        = <<-EOT
                    function getByEmail (email, callback) {
                  -   return callback(new Error("Whoops2!"))
                  +   return callback(new Error("Whoops!"))
                    }
                EOT
              - "login"           = <<-EOT
                    function login(email, password, callback) {
                      // This script should authenticate a user against the credentials stored in
                      // your database.
                      // It is executed when a user attempts to log in or immediately after signing
                      // up (as a verification that the user was successfully signed up).
                      //
                      // Everything returned by this script will be set as part of the user profile
                      // and will be visible by any of the tenant admins. Avoid adding attributes
                      // with values such as passwords, keys, secrets, etc.
                      //
                      // The `password` parameter of this function is in plain text. It must be
                      // hashed/salted to match whatever is stored in your database. For example:
                      //
                      //     var bcrypt = require('bcrypt@0.8.5');
                      //     bcrypt.compare(password, dbPasswordHash, function(err, res)) { ... }
                      //
                      // There are three ways this script can finish:
                      // 1. The user's credentials are valid. The returned user profile should be in
                      // the following format: https://auth0.com/docs/users/normalized/auth0/normalized-user-profile-schema
                      //     var profile = {
                      //       user_id: ..., // user_id is mandatory
                      //       email: ...,
                      //       [...]
                      //     };
                      //     callback(null, profile);
                      // 2. The user's credentials are invalid
                      //     callback(new WrongUsernameOrPasswordError(email, "my error message"));
                      //
                      //    Note: Passing no arguments or a falsey first argument to
                      //    `WrongUsernameOrPasswordError` will result in the error being logged as
                      //    an `fu` event (invalid username/email) with an empty string for a user_id.
                      //    Providing a truthy first argument will result in the error being logged
                      //    as an `fp` event (the user exists, but the password is invalid) with a
                      //    user_id value of "auth0|<first argument>". See the `Log Event Type Codes`
                      //    documentation for more information about these event types:
                      //    https://auth0.com/docs/deploy-monitor/logs/log-event-type-codes
                      // 3. Something went wrong while trying to reach your database
                      //     callback(new Error("my error message"));
                      //
                      // A list of Node.js modules which can be referenced is available here:
                      //
                      //    https://tehsis.github.io/webtaskio-canirequire/

                      const msg = 'Please implement the Login script for this database connection ' +
                        'at https://manage.auth0.com/#/connections/database';
                      return callback(new Error(msg));
                    }
                EOT -> null
              - "verify"          = <<-EOT
                    function verify(email, callback) {
                      // This script should mark the current user's email address as verified in
                      // your database.
                      // It is executed whenever a user clicks the verification link sent by email.
                      // These emails can be customized at https://manage.auth0.com/#/emails.
                      // It is safe to assume that the user's email already exists in your database,
                      // because verification emails, if enabled, are sent immediately after a
                      // successful signup.
                      //
                      // There are two ways that this script can finish:
                      // 1. The user's email was verified successfully
                      //     callback(null, true);
                      // 2. Something went wrong while trying to reach your database:
                      //     callback(new Error("my error message"));
                      //
                      // If an error is returned, it will be passed to the query string of the page
                      // where the user is being redirected to after clicking the verification link.
                      // For example, returning `callback(new Error("error"))` and redirecting to
                      // https://example.com would redirect to the following URL:
                      //     https://example.com?email=alice%40example.com&message=error&success=false

                      const msg = 'Please implement the Verify script for this database connection ' +
                        'at https://manage.auth0.com/#/connections/database';
                      return callback(new Error(msg));
                    }
                EOT -> null
            }
            # (24 unchanged attributes hidden)
```

## Environment 

- Terraform: v1.2.2
- Auth0 Provider: v0.30.3


We'll be closing this down but feel free to reopen if you have any other questions.

**Comment by Kos** [comment]
Thank you, I confirm it's fixed for me after updating to 0.31.0
